### ISSUE TITLE
Checking allocation failures

### ISSUE THREAD

**Issue by zherczeg** [PCRE2Project/pcre2#627]
Since a lot of effort spent on code stability somebody might want to work on allocation failure checking. This kind of testing rely on the custom allocators provided by PCRE2.

These should be the steps:

1) Pick a pattern, could be a random pattern

2) Compile / JIT compile the pattern
All allocations are successful, the number of allocations are counted. Their number is N

3) Loop:
```
for (I = 0; i < N; I++)
  Compile / JIT compile the pattern
  Only the first I allocations are successful, then the allocator returns with NULL
```

The engine should not crash. Furthermore, the custom free function should be called I times, and not with NULL argument. Memory checkers should be able to detect if some memory blocks are not freed.

This could be built into pcre2test.

**Comment by NWilson** [comment]
Yes, nice plan! We'd get a good coverage improvement too, since all those allocation failures are totally untested, at the moment.

I might get time to do it this week, or if not, then in January.
